### PR TITLE
Change deal frontend to call the deal backend api

### PR DESCRIPTION
**Pull request by limli**
Screenshot 2020-07-09 at 1 01 24 PM
Page will show loading animation while backend is being called

Screenshot 2020-07-09 at 1 00 11 PM
Page will display the deal once the data is fetched from the backend.
Restaurant says undefined because it is waiting for restaurant manager PR.
Upvote/downvote not implemented with backend yet.

Screenshot 2020-07-09 at 1 00 52 PM
If the deal id does not exist, then this will display instead

### CODE REVIEW

File: src/main/webapp/WEB-INF/jsp/deals.jsp [hunk 1]
```diff
@@ -6,71 +6,107 @@
...
+    }
+
+    #upvote-btn:hover {
+      color: #28a745;
+    }
+
+    #upvote-btn.active {
+      color: #28a745;
+    }
+  </style>
```
Comment (jyx11011) [inline src/main/webapp/WEB-INF/jsp/deals.jsp]: I think it's better to put them in a separate css file.

**Review by lisasgoh — DISMISSED** [review]

File: src/main/webapp/js/go-to-deal.js [hunk 2]
```diff
@@ -113,7 +84,72 @@ function createCommentElement(commentEntity) {
...
+        loadDealDataToPage(deal);
+      })
+      .fail(() => {
+        showNotFound();
+      });
+}
+
 addLoadEvent(() => {
-  getDeal(deal);
+  initDeal();
```
Comment (SuyashLakhotia) [inline src/main/webapp/js/go-to-deal.js:153]: Use the common function from `util.js` here?

File: src/main/java/com/google/step/servlets/DealDetailServlet.java [hunk 3]
```diff
@@ -46,13 +59,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    User poster = userManager.readUser(deal.posterId);
+
+    // TODO get real tags
+    List<Tag> tags = new ArrayList<>();
+
+    int votes = voteManager.getVotes(deal.id);
+
     response.setContentType("application/json;");
-    response.getWriter().println(JsonFormatter.getDealJson(deal));
+    response.getWriter().println(JsonFormatter.getDealJson(deal, restaurant, poster, tags, votes));
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/servlets/DealDetailServlet.java:79]: Should there be a test for these changes?

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/go-to-deal.js (hunk 2 above)
Comment (limli) [reply src/main/webapp/js/go-to-deal.js:153]: what common function? `addLoadEvent` is already from util

File: src/main/java/com/google/step/servlets/DealDetailServlet.java (hunk 3 above)
Comment (limli) [reply src/main/java/com/google/step/servlets/DealDetailServlet.java:79]: I added a test for JsonFormatter. The servlet test can be done in another PR.

**Review by SuyashLakhotia — APPROVED** [review]

**Pull request merged**